OpenSim/Examples/BuildDynamicWalker/BuildDynamicWalkerModel.cpp

```diff
@@ -0,0 +1,351 @@
+        rightKneeContactParams->addGeometry("PlatformContact");
+
+        HuntCrossleyForce* rightKneeForce = new HuntCrossleyForce(
+            rightKneeContactParams);
+        rightKneeForce->setName("RightKneeForce");
+        osimModel.addForce(rightKneeForce);
+
+        HuntCrossleyForce::ContactParameters *leftKneeContactParams =
+            new HuntCrossleyForce::ContactParameters(stiffness,
+                dissipation, staticFriction, dynamicFriction, viscosity);
+        leftKneeContactParams->addGeometry("LKneeContact");
+        leftKneeContactParams->addGeometry("PlatformContact");
+
+        HuntCrossleyForce* leftKneeForce = new HuntCrossleyForce(
+            leftKneeContactParams);
+        leftKneeForce->setName("LeftKneeForce");
+        osimModel.addForce(leftKneeForce);
+        
+        HuntCrossleyForce::ContactParameters *leftFootContactParams =
+            new HuntCrossleyForce::ContactParameters(stiffness,
+                dissipation, staticFriction, dynamicFriction, viscosity);
+        leftFootContactParams->addGeometry("LFootContact");
+        leftFootContactParams->addGeometry("PlatformContact");
+
+        HuntCrossleyForce* leftFootForce = new HuntCrossleyForce(
+            leftFootContactParams);
+        leftFootForce->setName("LeftFootForce");
+        osimModel.addForce(leftFootForce);
+
+        HuntCrossleyForce::ContactParameters *rightFootContactParams =
+            new HuntCrossleyForce::ContactParameters(stiffness,
+                dissipation, staticFriction, dynamicFriction, viscosity);
+        rightFootContactParams->addGeometry("RFootContact");
+        rightFootContactParams->addGeometry("PlatformContact");
+
+        HuntCrossleyForce* rightFootForce = new HuntCrossleyForce(
+            rightFootContactParams);
+        rightFootForce->setName("RightFootForce");
+        osimModel.addForce(rightFootForce);
+
+        CoordinateLimitForce* clfLHip = new CoordinateLimitForce("LHip_rz",
+            100, 1e6, -100, 1e6, 1e5, 5);
+        osimModel.addForce(clfLHip);
+        CoordinateLimitForce* clfRHip = new CoordinateLimitForce("RHip_rz",
+            100, 1e6, -100, 1e6, 1e5, 5);
+        osimModel.addForce(clfRHip);
+        CoordinateLimitForce* clfLKnee = new CoordinateLimitForce("LKnee_rz",
+            0, 1e6, -100, 1e6, 1e5, 5);
+        osimModel.addForce(clfLKnee);
+        CoordinateLimitForce* clfRKnee = new CoordinateLimitForce("RKnee_rz",
+            0, 1e6, -100, 1e6, 1e5, 5);
+        osimModel.addForce(clfRKnee);
+
+        // Save the model to a file
+        osimModel.print("DynamicWalkerModel.osim");
+    } catch (const std::exception& ex) {
+        std::cout << ex.what() << std::endl;
+        return 1;
+    } catch (...) {
+        std::cout << "UNRECOGNIZED EXCEPTION" << std::endl;
+        return 1;
+    }
+    std::cout << "OpenSim example completed successfully" << std::endl;
+    // std::cout << "Press return to continue" << std::endl;
```

Comment:
Yes; so that the tests don't time-out.

Reply:
What are we assuming about the knowledge of the target audience? Possible thoughts:

- The code was commented out because it doesn't work.
- Uncommenting the code causes something to break.
- The code was commented out and should have been removed.

Regardless, I recommend leaving it as it is.

Reply:
This bit of code is useful when running command-line examples on Windows so that the command window doesn't close on you abruptly, so I left it in. In other places, we've added a way to detect if the code is being run as part of a test via an environment variable, but I decided not to spend time on that this time. I agree with your final suggestion.